io.github.benjamimgois.goverlay.yml

```diff
@@ -0,0 +1,283 @@
+    add-ld-path: lib
+    no-autodownload: false
+    autodelete: false
+
+  org.freedesktop.Platform.VulkanLayer.vkBasalt:
+    directory: lib/extensions/vulkan/vkBasalt
+    version: stable
+    add-ld-path: lib
+    no-autodownload: false
+    autodelete: false
+
+finish-args:
+  # GPU access for Vulkan/OpenGL overlays
+  - --device=dri
+
+  # Filesystem access for configuration files
+  - --filesystem=xdg-config/MangoHud:create
+  - --filesystem=xdg-config/vkBasalt:create
+  - --filesystem=xdg-config/goverlay:create
+  - --filesystem=xdg-data/goverlay:create
+  - --filesystem=home
+
+  # Read-only access to /sys for hardware detection
+  - --filesystem=/sys/class/drm:ro
+  - --filesystem=/sys/bus/pci:ro
+  - --filesystem=/sys/class/net:ro
+
+  # Network access for downloads (OptiScaler, ReShade shaders, updates)
+  - --share=network
+
+  # GUI requirements
+  - --share=ipc
+  - --socket=wayland
+  - --socket=fallback-x11
+
+  # Desktop notifications
+  - --talk-name=org.freedesktop.Notifications
+
+cleanup:
+  - /include
+  - /lib/pkgconfig
+  - /share/man
+  - '*.la'
+
+modules:
+  # Free Pascal Compiler (FPC) - Binary distribution
+  - name: fpc
+    buildsystem: simple
+    build-commands:
+      - |
+        # Extract the appropriate FPC tarball based on architecture
+        if [ -f fpc-3.2.2.x86_64-linux.tar ]; then
+          tar xf fpc-3.2.2.x86_64-linux.tar
+          cd fpc-3.2.2.x86_64-linux
+        elif [ -f fpc-3.2.2.aarch64-linux.tar ]; then
+          tar xf fpc-3.2.2.aarch64-linux.tar
+          cd fpc-3.2.2.aarch64-linux
+        fi
+        echo "/app" | ./install.sh
+      - |
+        # Create proper fpc.cfg in /app/etc
+        mkdir -p /app/etc
+        /app/lib/fpc/3.2.2/samplecfg /app/lib/fpc/3.2.2 /app/etc
+      - |
+        # Add library path to fpc.cfg
+        echo "-Fl/app/lib" >> /app/etc/fpc.cfg
+    sources:
+      - type: file
+        url: https://sourceforge.net/projects/freepascal/files/Linux/3.2.2/fpc-3.2.2.x86_64-linux.tar/download
+        sha256: 5adac308a5534b6a76446d8311fc340747cbb7edeaacfe6b651493ff3fe31e83
+        dest-filename: fpc-3.2.2.x86_64-linux.tar
+        only-arches:
+          - x86_64
+        x-checker-data:
+          type: html
+          url: https://sourceforge.net/projects/freepascal/files/Linux/
+          pattern: fpc-(\d+\.\d+\.\d+)\.x86_64-linux\.tar
+      - type: file
+        url: https://sourceforge.net/projects/freepascal/files/Linux/3.2.2/fpc-3.2.2.aarch64-linux.tar/download
+        sha256: dbe159e6f5e694b88ff8e9e60e1c300b0caa353628e205728ce7f20703663887
+        dest-filename: fpc-3.2.2.aarch64-linux.tar
+        only-arches:
+          - aarch64
+
+  # Lazarus IDE (provides lazbuild)
+  - name: lazarus
+    buildsystem: simple
+    build-commands:
+      - |
+        export FPCDIR=/app/lib/fpc/3.2.2
+        export PATH=/app/bin:$PATH
+        export Qt6Pas_LibDir=/app/lib
+        # Detect architecture for FPC units path
+        FPC_ARCH=$(uname -m)
```

Comment:
use $FLATPAK_ARCH no need to uname

Reply:
lazarus is also in the freepascal SDK. There is no need to build either.